object-store/src/main/java/org/hypertrace/config/objectstore/ScopeResolutionIdentifiedObjectStoreWithFilter.java

```diff
@@ -0,0 +1,138 @@
+package org.hypertrace.config.objectstore;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.hypertrace.config.service.change.event.api.ConfigChangeEventGenerator;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+/**
+ * ScopeResolutionIdentifiedObjectStoreWithFilter is an abstraction over {@link
+ * IdentifiedObjectStoreWithFilter} to allow scope resolution of fetched configs
+ *
+ * @param <T> Config Object
+ * @param <F> Filter
+ * @param <S> Scope
+ */
+public abstract class ScopeResolutionIdentifiedObjectStoreWithFilter<T extends Message, F, S>
+    extends IdentifiedObjectStoreWithFilter<T, F> {
+
+  protected final T defaultConfig;
+
+  protected ScopeResolutionIdentifiedObjectStoreWithFilter(
+      ConfigServiceGrpc.ConfigServiceBlockingStub configServiceBlockingStub,
+      String resourceNamespace,
+      String resourceName,
+      ConfigChangeEventGenerator configChangeEventGenerator,
+      T defaultConfig) {
+    super(configServiceBlockingStub, resourceNamespace, resourceName, configChangeEventGenerator);
+    this.defaultConfig = defaultConfig;
+  }
+
+  public T getDefaultConfig() {
+    return defaultConfig;
+  }
+
+  public List<ContextualConfigObject<T>> getAllFilteredResolvedObjects(
+      RequestContext context, F filter) {
+    Map<S, ContextualConfigObject<T>> configObjectsMap = getConfigObjectsMap(context, filter);
+    return configObjectsMap.values().stream()
+        .map(
+            configObject ->
+                ContextualConfigObjectImpl.updateData(
+                    configObject,
+                    resolveConfig(extractScope(configObject.getData()), configObjectsMap)))
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public List<T> getAllFilteredResolvedConfigData(RequestContext context, F filter) {
+    return getAllFilteredResolvedObjects(context, filter).stream()
+        .map(ConfigObject::getData)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public Optional<ContextualConfigObject<T>> getFilteredResolvedObject(
+      RequestContext context, F filter, S scope) {
+    Map<S, ContextualConfigObject<T>> configObjectsMap = getConfigObjectsMap(context, filter);
+    return Optional.ofNullable(configObjectsMap.get(scope))
```

Comment:
if there is no config stored for A3, configStore has no idea of existence of api A3.. It's only going to return the resolved version of the configs stored..

Reply:
oh it will return that in this impl too - I misunderstood ur initial question.. what I explained was the impl for `getAll`.. 
if a call is made for getResolved for A3 and its config is not present, this impl should surely return config for S1 - forgot to use the chain call.. will fix..

Reply:
update - just checked.. the impl will return S1 - the `resolveConfig` method calls the chain

Reply:
Maybe I'm looking at old code, but we don't call `resolveConfig` unless the requested scope is in the map. If it's a scope no persisted config has, then we'll always return empty

```java
Optional.ofNullable(configObjectsMap.get(scope))
```

Reply:
yes old code - had missed it in 1 method, now fixed..

Reply:
Maybe a missed push? Looks like the latest still has this:
https://github.com/hypertrace/config-service/pull/134/files#diff-beb6243dae76a59024110378831901720eda9039152e42b1bde6f1ff615d2957R51-R53

Reply:
yeah - fixed..